File: connectors-editors/mantis-editor/src/main/java/com/taskadapter/connector/mantis/editor/OtherMantisFieldsPanel.java
package com.taskadapter.connector.mantis.editor;

import com.taskadapter.connector.mantis.MantisConfig;
import com.taskadapter.web.configeditor.EditorUtil;
import com.vaadin.flow.component.checkbox.Checkbox;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.data.binder.Binder;

public class OtherMantisFieldsPanel extends VerticalLayout {
    private static String DEFAULT_PANEL_CAPTION = "Additional Info";

    private final Binder<MantisConfig> binder;

    public OtherMantisFieldsPanel(Binder<MantisConfig> binder) {
        this.binder = binder;
        buildUi();
    }

    private void buildUi() {
        setSpacing(true);

        Checkbox findUserByName = EditorUtil.checkbox("Find users based on assignee's name",
                "This option can be useful when you need to export a new MSP project file to Redmine/JIRA/MantisBT/....\n"
                        + "Task Adapter can load the system's users by resource names specified in the MSP file\n"
                        + "and assign the new tasks to them.\n"
                        + "Note: this operation usually requires 'Admin' permission in the system.",
                binder, "findUserByName");


        add(
                new Label(DEFAULT_PANEL_CAPTION),
                findUserByName
        );
    }

}

File: connectors-editors/mantis-editor/src/main/scala/com/taskadapter/connector/mantis/editor/OtherMantisFieldsPanel.scala
package com.taskadapter.connector.mantis.editor

import com.taskadapter.connector.mantis.MantisConfig
import com.taskadapter.vaadin14shim.VerticalLayout
import com.taskadapter.web.configeditor.Editors
import com.vaadin.data.util.MethodProperty
import com.vaadin.ui.Panel

class OtherMantisFieldsPanel(val config: MantisConfig) extends Panel {
  private val DEFAULT_PANEL_CAPTION = "Additional Info"

  val verticalLayout = new VerticalLayout

  setCaption(DEFAULT_PANEL_CAPTION)
  setContent(verticalLayout)
  verticalLayout.setSpacing(true)
  verticalLayout.add(Editors.createFindUsersElement(
    new MethodProperty[java.lang.Boolean](config, "findUserByName")))
}